src/OVAL/oval_session.c

```diff
@@ -0,0 +1,463 @@
+/*
+ * Copyright 2015 Red Hat Inc., Durham, North Carolina.
+ * All Rights Reserved.
+ *
+ * This library is free software; you can redistribute it and/or
+ * modify it under the terms of the GNU Lesser General Public
+ * License as published by the Free Software Foundation; either
+ * version 2.1 of the License, or (at your option) any later version.
+ *
+ * This library is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the GNU
+ * Lesser General Public License for more details.
+ *
+ * You should have received a copy of the GNU Lesser General Public
+ * License along with this library; if not, write to the Free Software
+ * Foundation, Inc., 59 Temple Place, Suite 330, Boston, MA  02111-1307  USA
+ *
+ * Author:
+ * 		Michal Šrubař <msrubar@redhat.com>
+ */
+
+#ifdef HAVE_CONFIG_H
+#include <config.h>
+#endif
+
+#include <libgen.h>
+#include <unistd.h>
+#include <string.h>
+
+#include "common/alloc.h"
+#include "common/util.h"
+#include "common/_error.h"
+#include "common/oscapxml.h"
+#include "source/xslt_priv.h"
+#include "public/oval_agent_api.h"
+#include "public/oval_session.h"
+#include "../DS/public/ds_sds_session.h"
+#include "oscap_source.h"
+
+static const char *oscap_productname = "cpe:/a:open-scap:oscap";
```

Comment:
The product name itself is common and used in both xccdf and oval sessions. Maybe it would be usefull to put it into the ```oscap.h``` but lots of functions is DEPRECATED in there. Do we plan to get rid of it in the future? The string ```"cpe:/a:open-scap:oscap"``` is already defined as macro in ```oscap-tool.h```